gateway/gateway.go

```diff
@@ -27,14 +31,74 @@ type gateway struct {
 	mux               *http.ServeMux
 }
 
+// CollectionOperationsInterceptor parse collection operators and stores in corresponding message fields
+func CollectionOperationsInterceptor(parentCtx context.Context, method string, req, reply interface{}, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
```

Comment:
That is client interceptor on  gateway, which parse  URL parameters and store in corresponding collection fields.